src/git-kill

```diff
@@ -12,31 +12,47 @@ function branch_to_kill {
 
 
 function preconditions {
-  target_branch=$(branch_to_kill "$@")
+  target_branch_name=$(branch_to_kill "$@")
 
-  ensure_is_feature_branch "$target_branch" "You can only kill feature branches."
+  ensure_is_feature_branch "$target_branch_name" "You can only kill feature branches."
 
   if [ "$HAS_REMOTE" = true ]; then
       fetch
   fi
 
-  if [ "$target_branch" != "$INITIAL_BRANCH_NAME" ]; then
-    ensure_has_branch "$target_branch"
+  if [ "$target_branch_name" != "$INITIAL_BRANCH_NAME" ]; then
+    ensure_has_branch "$target_branch_name"
+  fi
+}
+
+
+function set_previous_branch {
+  # previous branch was deleted
+  if [ "$target_branch_name" == "$INITIAL_PREVIOUS_BRANCH_NAME" ]; then
```

Comment:
Based on your description, I think the new "previous branch" is always the existing previous branch, except if that branch gets deleted by the current Git Town command. In this case its always main. 

If I understand this correctly, you could automate this fully, without customizations in each Git Town command.